### PR TITLE
Preparations for VSUP migration to Ansible

### PR DESCRIPTION
**Pull request by xkostka2**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -455,10 +456,17 @@ perun_engine_mounts_additional: []
...
+  - RewriteRule ^[/]?$ https://%{HTTP_HOST}/fed/gui/ [L]
+# whether enable redirect from / to something else
+perun_apache_default_registrar_enabled: yes
+perun_apache_default_registrar_rewrite_rules: ""
 # whether to enable redirect from / to /{{ perun_apache_profile_default_auth }}/profile/
 perun_apache_default_profile_enabled: no
 # whether to activate CSRF protection in Perun RPC on main hostname
 perun_apache_csrf_protection_enabled: no
+# whether to add CORS headers by Apache
+perun_apache_cors_headers_enabled: yes
```
Comment (martin-kuba) [inline defaults/main.yml:469]: the variable was removed, because it was always true
```suggestion
```

File: templates/perun.properties.j2 [hunk 2]
```diff
@@ -7,7 +7,7 @@ perun.admins = {{ (perun_rpc_admins+perun_rpc_additional_admins)|join(', ') }}
 perun.engine.principals = {{ perun_rpc_engine_principals|join(', ') }}
 
 # Principals for the Registrar
-perun.registrar.principals =
+perun.registrar.principals = {{ perun_registrar_principals|join(', ') }}
```
Comment (martin-kuba) [inline templates/perun.properties.j2:10]: keep the property as is

File: templates/sites-enabled/perun-api.conf.j2 [hunk 3]
```diff
@@ -180,5 +180,5 @@
         Authtype oauth20
         Require valid-user
     </LocationMatch>
-
+{{ perun_apache_special_config3 }}
```
Comment (martin-kuba) [inline templates/sites-enabled/perun-api.conf.j2:183]: rename it to something that shows that it is used for API, like perun_apache_api_special_config
```suggestion
{{ perun_apache_api_special_config }}
```

**Review by martin-kuba — CHANGES_REQUESTED** [review]

File: templates/instance_configs/instanceConfig.json.j2 [hunk 4]
```diff
@@ -45,5 +45,6 @@
   "logo": {{ perun_ngui_logo|to_json }},
   "theme": {{ perun_ngui_theme|to_nice_json(indent=2,ensure_ascii=False)|indent(2) }},
   "brandings": {{ perun_ngui_admin_brandings|to_nice_json(indent=2,ensure_ascii=False)|indent(2) }},
-  "enforce_consents": {{ perun_ngui_admin_enforce_consents|bool|to_json }}
+  "enforce_consents": {{ perun_ngui_admin_enforce_consents|bool|to_json }},
+  "profile_label_en": "Profile",
```
Comment (zlamalp) [inline templates/instance_configs/instanceConfig.json.j2:49]: `profile_label_en` is part of `defaultConfig.json` so I don't think we need to re-define it here unless custom value is expected.

File: defaults/main.yml [hunk 5]
```diff
@@ -454,6 +455,11 @@ perun_engine_mounts_additional: []
 perun_apache_in_maintenance: no
 # whether to enable redirect from / to /fed/gui/
 perun_apache_default_gui_enabled: yes
+perun_apache_default_gui_rewrite_rules:
+  - RewriteRule ^[/]?$ https://%{HTTP_HOST}/fed/gui/ [L]
+# whether enable redirect from / to something else
+perun_apache_default_registrar_enabled: yes
+perun_apache_default_registrar_rewrite_rules: ""
```
Comment (zlamalp) [inline defaults/main.yml:462]: This should be revisited based on the feedback provided about the template for apache config.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: templates/perun-engine.j2 [hunk 6]
```diff
@@ -1,6 +1,9 @@
 {{ ansible_managed | comment }}
 
 # Allow perun-engine to call Perun
+{% if perun_engine_lwp_ssl_verify_hostname_flag is defined %}
```
Comment (martin-kuba) [inline templates/perun-engine.j2:4]: It is not clear form the name of the variable what the values mean.
```suggestion
{% if perun_engine_verify_hostname_disabled is defined and perun_engine_verify_hostname_disabled %}
```

**Review by martin-kuba — CHANGES_REQUESTED** [review]

File: templates/sites-enabled/perun.conf.j2 [hunk 7]
```diff
@@ -144,7 +144,7 @@ ShibCompatValidUser on
   ####################################
   ##     Default GUI              ####
   ####################################
-  RewriteRule ^[/]?$ https://%{HTTP_HOST}/fed/gui/ [L]
+  RewriteRule ^[/]?$ https://%{HTTP_HOST}/{{ perun_apache_gui_default_auth }}/gui/ [L]
```
Comment (zlamalp) [inline templates/sites-enabled/perun.conf.j2:147]: You should do the same on line 170 (generic redirect for the GUI) as this line apply only to case when GUI is a default app on the domain.

File: defaults/main.yml [hunk 8]
```diff
@@ -454,6 +455,7 @@ perun_engine_mounts_additional: []
 perun_apache_in_maintenance: no
 # whether to enable redirect from / to /fed/gui/
 perun_apache_default_gui_enabled: yes
+perun_apache_gui_default_auth: "fed"
```
Comment (zlamalp) [inline defaults/main.yml:458]: Probably move it closer to the line 510, where a similar config for registrar is placed. Also probably update the comment on 456 where /fed/gui/ should mention configurable prefix like the comment on 459.

**Review by zlamalp — CHANGES_REQUESTED** [review]

**Review by zlamalp — APPROVED** [review]

**Review by martin-kuba — APPROVED** [review]

**Pull request merged**